sky/cli.py

```diff
@@ -3985,6 +4052,449 @@ def spot_dashboard(port: Optional[int]):
+    if all:
+        service_names = [
+            record['name'] for record in global_user_state.get_services()
+        ]
+    else:
+        service_names = _get_glob_services(service_names)
+
+    if not service_names:
+        click.echo('\nService(s) not found (tip: see `sky serve status`).')
+        return
+
+    plural = '' if len(service_names) == 1 else 's'
+    if not yes:
+        service_name_list = ', '.join(service_names)
+        click.confirm(
+            f'Tearing down {len(service_names)} service{plural}: '
+            f'{service_name_list}. Proceed?',
+            default=True,
+            abort=True,
+            show_default=True)
+
+    progress = rich_progress.Progress(transient=True,
+                                      redirect_stdout=False,
+                                      redirect_stderr=False)
+    task = progress.add_task(
+        f'[bold cyan]Tearing down {len(service_names)} service{plural}[/]',
+        total=len(service_names))
+
+    def _down_service(name: str):
+        success_progress = False
+        try:
+            sky.serve_down(name, purge)
+        except RuntimeError as e:
+            message = (
+                f'{colorama.Fore.RED}Tearing down service {name!r}...failed. '
+                'Please manually clean up the replicas and then use --purge '
+                f'to clean up the controller.{colorama.Style.RESET_ALL}'
+                f'\nReason: {common_utils.format_exception(e)}.')
+        else:
+            message = (
+                f'{colorama.Fore.GREEN}Tearing down service {name!r}...done.'
+                f'{colorama.Style.RESET_ALL}')
+            success_progress = True
+
+        progress.stop()
+        click.echo(message)
+        if success_progress:
+            progress.update(task, advance=1)
+        progress.start()
+
+    with progress:
+        subprocess_utils.run_in_parallel(_down_service, service_names)
+        progress.live.transient = False
+        # Make sure the progress bar not mess up the terminal.
+        progress.refresh()
+
+
+@serve.command('logs', cls=_DocumentedCodeCommand)
+@click.option(
+    '--follow/--no-follow',
+    is_flag=True,
+    default=True,
+    help=('Follow the logs of the job. [default: --follow] '
+          'If --no-follow is specified, print the log so far and exit.'))
+@click.option('--controller',
+              is_flag=True,
+              default=False,
+              required=False,
+              help='Show the controller logs of this service.')
+@click.option('--load-balancer',
+              is_flag=True,
+              default=False,
+              required=False,
+              help='Show the load balancer logs of this service.')
+@click.option('--target',
+              default=None,
+              type=click.Choice(['controller', 'load-balancer', 'replica'],
+                                case_sensitive=False),
+              required=False,
+              help='Target to stream logs.')
```

Comment:
A bit surprising to see this given other flags. Any reason to have this?

Reply:
It was originally Zhanghao's idea, to align CLI API with Python API. See [here](https://github.com/skypilot-org/skypilot/pull/2458#discussion_r1347781252) for more information